python/cudf/cudf/tests/test_json.py

```diff
@@ -327,3 +327,25 @@ def test_json_bad_protocol_string():
     got = cudf.read_json(test_string, lines=True)
 
     assert_eq(expect, got)
+
+
+def test_json_corner_case_with_escape_and_double_quote_char(tmpdir):
+    fname = tmpdir.mkdir("gdf_json").join("tmp_json_escape_double_quote")
+
+    pdf = pd.DataFrame(
+        {
+            "a": ['ab"cd', "?1sde", "rdsd", "halt"],
+            "b": ["234", "\\", '\\"', "type"],
+            "c": ["aeiou", "try", "json", "cudf"],
```

Comment:
I think there's a few more cases we can add to this test, now that the escape state is robust, something like:
`"\t", "\t\\", "\\\t", 'a\tb\t"'`